src/sonic-yang-models/yang-models/sonic-types.yang

```diff
@@ -114,4 +114,28 @@ module sonic-types {
             pattern "percentage|used|free|PERCENTAGE|USED|FREE";
         }
     }
+
+    typedef interface_type {
```

Comment:
Again, if possible, point to the HLD which shows from where these values are coming? thanks

Reply:
https://github.com/Junchao-Mellanox/SONiC/blob/port-auto-neg/doc/port_auto_neg/port-auto-negotiation-design.md#config-interface-type